analysis/descriptives/venn.R

```diff
@@ -138,55 +142,56 @@ for (outcome in outcomes) {
     print('Record the number contributing to each source combination')
     
     df[nrow(df)+1,] <- c(outcome,
-                         only_snomed = nrow(tmp %>% filter(snomed_contributing==T)),
-                         only_hes = nrow(tmp %>% filter(hes_contributing==T)),
-                         only_death = nrow(tmp %>% filter(death_contributing==T)),
-                         snomed_hes = nrow(tmp %>% filter(snomed_hes_contributing==T)),
-                         snomed_death = nrow(tmp %>% filter(snomed_death_contributing==T)),
-                         hes_death = nrow(tmp %>% filter(hes_death_contributing==T)),
-                         snomed_hes_death = nrow(tmp %>% filter(snomed_hes_death_contributing==T)),
-                         total_snomed = nrow(tmp %>% filter(!is.na(snomed))),
-                         total_hes = nrow(tmp %>% filter(!is.na(hes))),
-                         total_death = nrow(tmp %>% filter(!is.na(death))),
-                         total = nrow(tmp),
+                         only_snomed_midpoint6 = nrow(tmp %>% filter(snomed_contributing==T)),
+                         only_hes_midpoint6 = nrow(tmp %>% filter(hes_contributing==T)),
+                         only_death_midpoint6 = nrow(tmp %>% filter(death_contributing==T)),
+                         snomed_he_midpoint6s = nrow(tmp %>% filter(snomed_hes_contributing==T)),
+                         snomed_death_midpoint6 = nrow(tmp %>% filter(snomed_death_contributing==T)),
+                         hes_death_midpoint6 = nrow(tmp %>% filter(hes_death_contributing==T)),
+                         snomed_hes_death_midpoint6 = nrow(tmp %>% filter(snomed_hes_death_contributing==T)),
+                         total_snomed_midpoint6 = nrow(tmp %>% filter(!is.na(snomed))),
+                         total_hes_midpoint6 = nrow(tmp %>% filter(!is.na(hes))),
+                         total_death_midpoint6 = nrow(tmp %>% filter(!is.na(death))),
+                         total_midpoint6_derived = nrow(tmp),
                          error = "")
     
     # Fix source contribution for any dementia outcome -------------------------
+    print("Fix source contribution for any dementia")
```

Comment:
Tested 5 times before open the PR. Tested twice before and after moving the lines outside loop. In the 7 test, only the outcomes with one source showed NAs for the other columns.
Columns renamed after the output redaction line.

Reply:
Okay, this is working for me now in terms of populating the table with numbers but the approach is not suitable for two reasons:

(1) Individuals can be in more than one outcome group. For instance, a person might be in alzheimer_disease and vascular_dementia. If that is the case, summing the already summarised information about these outcomes will double count this individual. 

(2) Any dementia is defined as the combination of alzheimer_disease, vascular_dementia, lewy_body_dementia, other_dementias, and unspecified_dementias but the Venn table only contains information on alzheimer_disease, vascular_dementia and lewy_body_dementia so you are not considering all contributing diagnoses.

Reply:
1. Originally there was a specific variable for any_dementia, but you and Rachel decided to use the one that combines alzheimer_disease, vascular_dementia, lewy_body_dementia, other_dementias, and unspecified_dementias (the original study definitions did not included lewy body). 

The any dementia variable already has individuals that may be in both alzheimer_disease and vascular_dementia, that's how the variable was built. The original approach, a specific any_dementia variable, did not have this issue.  You and Rachel wanted higher number of patients.

2. The missing unspecified dementia can be fixed by adding the model, but such variable is not in the protocol and it will  not completely address point 1.

Reply:
(1) Individuals being in more than one outcome group is not a problem, it is expected. However, it does need to be accounted for when calculating the source of any_dementia for the Venn diagrams because only the earliest of these diagnoses is used to date the any_dementia record.

(2) You do not need to add a model for unspecified dementia to access this information. The information you require from the model input is the date of any dementia. `venn_*.rds` contains the contributing variables, including `tmp_out_date_unspecified_dementias_*`, that you require.

The solution to both issues above is to perform the calculations for any_dementia from the raw data (as you do for the other forms of dementia) accounting for the fact that any_dementia has more contributing variables.

Reply:
This is outside of the redaction format, so I will queue it, and addrees it after table 1.

Reply:
Okay. This seems sensible. I have recorded the issue here: https://github.com/opensafely/post-covid-neurodegenerative/issues/73.